test/web-api.test.js

```diff
@@ -157,3 +158,51 @@ test('Error when Response.bodyUsed', async (t) => {
   const body = response.json()
   t.equal(body.code, 'FST_ERR_REP_RESPONSE_BODY_CONSUMED')
 })
+
+test('allow to pipe with fetch', async (t) => {
+  t.plan(2)
+
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+
+  fastify.get('/', function (request, reply) {
+    return fetch(`${fastify.listeningOrigin}/fetch`, {
+      method: 'GET'
+    })
+  })
+
+  fastify.get('/fetch', function (request, reply) {
+    reply.code(200).send({ ok: true })
+  })
+
+  await fastify.listen()
+
+  const response = await fastify.inject({ method: 'GET', path: '/' })
+
+  t.equal(response.statusCode, 200)
+  t.same(response.json(), { ok: true })
+})
+
+test('allow to pipe with undici.fetch', async (t) => {
```

Comment:
@gurgunday
It is why I use `Object.prototype.toString.call(payload) === '[object Response]'`.